backend/danswer/danswerbot/slack/handlers/handle_buttons.py

```
@@ -40,6 +40,75 @@
 logger_base = setup_logger()
 
 
+# Constants for curated response handling
+CURATED_RESPONSE_CONFIG_KEY = "curated_response_config"
+ENABLE_CURATED_RESPONSE_KEY = "enable_curated_response_integration"
+RESPONSE_MESSAGE_KEY = "response_message"
+USER_TITLE_FILTER_KEY = "curated_response_user_title_filter"
+USER_KEY = "user"
+USER_ID_KEY = "id"
+USER_PROFILE_KEY = "profile"
+USER_TITLE_KEY = "title"
+DEFAULT_CURATED_RESPONSE_MESSAGE = "contact customer support for more help"
+
+
+def handle_curated_response(
+    slack_bot_config: Any,
+    client: SocketModeClient,
+    req: SocketModeRequest,
+    channel_id: str,
+    thread_ts: str,
+) -> bool:
+    """Handle curated response based on user title filter.
+
+    Returns:
+        bool: True if a curated response was sent, else returns False.
+    """
+    if not slack_bot_config or not slack_bot_config.channel_config:
+        return False
+
+    channel_conf = slack_bot_config.channel_config
+    curated_response_config = channel_conf.get(CURATED_RESPONSE_CONFIG_KEY, {})
+
+    # Early return if curated response is not enabled
+    if not curated_response_config.get(ENABLE_CURATED_RESPONSE_KEY, False):
+        return False
+
+    user_title_filter = channel_conf.get(USER_TITLE_FILTER_KEY, [])
+    sender_id = req.payload.get(USER_KEY, {}).get(USER_ID_KEY)
+
+    if not user_title_filter or not sender_id:
+        return False
+
+    try:
+        user_info = client.web_client.users_info(user=sender_id)
+        user_data = user_info.get(USER_KEY)
+        if not user_data:
+            return False
+
+        user_profile = user_data.get(USER_PROFILE_KEY, {})
+        user_title = user_profile.get(USER_TITLE_KEY, "").lower()
+
+        # Check if user title matches any in the filter list
+        if user_title in [title.lower() for title in user_title_filter]:
+            response_message = curated_response_config.get(
+                RESPONSE_MESSAGE_KEY, DEFAULT_CURATED_RESPONSE_MESSAGE
+            )
+
```

`RESPONSE_MESSAGE_KEY` is a required field which would always be set right? When would the case arise for falling back to the default message?

---

Yes, but ideally we should always have a fallback

---

It would make sense if it was an optional field. 
If it is truly required, silently falling back could hide a misconfigured `curated_response_config`?
Shoudn't we validate and fail fast, to keep the system predictable?

---

Okay let me remove this and fetch only the required field

---

done and also it fails fast now